lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -542,7 +542,7 @@ def add_contact(doc, fullname, options)
               doc["v1"].nr billing_address[:phone].gsub(/\D/, '') if billing_address[:phone]
             end
             doc["v1"].addrLn1 billing_address[:address1]
-            doc["v1"].addrLn2 billing_address[:address2]
+            doc["v1"].addrLn2 billing_address[:address2] unless billing_address[:address2].blank?
```

Comment:
@nfarve I know there are a couple places in Active Merchant where we currently use the `.blank?` method, but we're trying to slowly remove those/not continue using it since it's actually an [Active Support](https://github.com/rails/rails/tree/master/activesupport) extension and we'd like to remove our dependency on Active Support. It's current usage is a holdover from when Active Merchant was more tied to being used in a Rails application.

That being said, it's not well documented that we're no longer using that convention and instead going with something like:

`doc["v1"].addrLn2 billing_address[:address2] if billing_address[:address2]`

Apologies on the poor documentation! We're not terribly clear about this in general and it's (unfortunately) become more tribal knowledge. Likewise, looking at the remote test we should probably change the `options` hash to instead have:

```ruby
options[:shipping_address][:address2] = nil
options[:billing_address][:address2] = nil
```

Since in general someone will probably not add those keys to a hash to begin with so accessing that key in the hash will just return a `nil`.